scripts/eofConvention.fsx

```diff
@@ -10,15 +12,21 @@ let EolAtEof(fileInfo: FileInfo) =
     else
         true
 
-let NotInDir (dirName: string) (fileInfo: FileInfo) =
-    not (fileInfo.FullName.Contains $"%c{Path.DirectorySeparatorChar}%s{dirName}%c{Path.DirectorySeparatorChar}")
+let InDir (dirName: string) (fileInfo: FileInfo) = 
+    fileInfo.FullName.Contains $"%c{Path.DirectorySeparatorChar}%s{dirName}%c{Path.DirectorySeparatorChar}"
+
+let NotInDirs (dirNames: List<string>) (fileInfo: FileInfo) =
+    not (dirNames 
+        |> Seq.map (fun dirName -> InDir dirName fileInfo)
+        |> Seq.contains true)
+
+let whitelistFolders = ["node_modules"; ".git"; "Debug"; "obj"; "bin"; "DummyFiles"]
 
 let invalidFiles = 
     Directory.GetFiles(".", "*.*", SearchOption.AllDirectories) 
     |> Seq.map (fun pathStr -> FileInfo pathStr)
-    |> Seq.filter (NotInDir "node_modules")
-    |> Seq.filter (NotInDir ".git")
-    |> Seq.filter (NotInDir "DummyFiles")
+    |> Seq.filter (NotInDirs whitelistFolders)
+    |> Seq.filter (fun fileInfo -> not (FileConventions.HasBinaryContent fileInfo))
```

Comment:
@realmarv sorry, wrong order of commits; this commit that moves the logic from the .fsx file to FileConventions library has to be done before you write any unit test, before you fix the bug! otherwise the diff for the bugfix is not clear

Reply:
Shouldn't I first implement HasBinaryContent function and then call it in the script?

Reply:
yes but we need the first commits (before creating any test) to move the logic from script to library; otherwise the moving of the code from one place to another is obfuscating the fix

Reply:
Done